sections/sprint-4/updated-uppaal.tex

```diff
@@ -5,6 +5,47 @@ \section{Updating the \uppaal model}
+
+First of we are going to look at the simplified model to get a general overview of the flow of the system.
+There are two templates in the simplified model which are Host and Client.
+
+\subsection*{Simplified host}
+The simplified host can be seen on \autoref{fig:sprint4-uppaal-simplified-host}. 
+The \uppProc{Host} starts in the \uppLoc{Lobby} and waits for all Clients to connect.
+Everytime a \uppProc{Client} connects the \uppProc{Host} transitions to  \uppLoc{Connecting}.
+It then either acknowledges the connection or waits for the  \uppProc{Client} to timeout.
+When the connection is acknowledged it transitions to \uppLoc{Connected} and then sends configurations to the \uppProc{Client} and transitions back to the lobby.
+When all clients have connected the \uppProc{Host} transitions to \uppLoc{InGame} and signals to all of the connected Clients that the game is starting through \uppOut{TCP_start_game}.
+InGame the \uppProc{Host} can either send tag position through UDP, send goal postion and goal scored through TCP or end the game through TCP and transition back to \uppLoc{Lobby}.
+\begin{figure}[h]
+    \centering
+    \includegraphics[width=1\linewidth]{sprint4/simplified-host.png}
+    \caption{The Host template of the simplified \uppaal model.}
+    \label{fig:sprint4-uppaal-simplified-host}
+\end{figure}
+
+\subsection*{Simplified Client}
+The simplified Client can be seen on  \autoref{fig:sprint4-uppaal-simplified-client}.
+The \uppProc{Client} starts in the \uppLoc{Initial} it then tries to connect and transitions to \uppLoc{Connecting}.
+While in \uppLoc{Connecting} it waits for a acknowledgement from the \uppProc{Host}.
+If it does not recieve a acknowledgement before the timeout limit it outputs a \uppOut{TCP_timeout} and transitions back to \uppLoc{Initial}.
+Otherwise if the \uppProc{Client} recieves a acknowledgement from the \uppProc{Host} it transitions to \uppLoc{Setup} and waits for the \uppProc{Host} to send configurations.
+When the \uppProc{Client} recieves the configurations it transitions to \uppLoc{Lobby} and then waits for the game to start.
+When the \uppProc{Host} starts the game, all \uppProc{Clients} that are in \uppLoc{Lobby} transition to \uppLoc{InGame}.
+From there the \uppProc{Client} waits for the \uppProc{Host} to send messages.
+It can recieve a TCP message or a UDP message which makes it transition to InGame again or it can recieve a message that the game has ended which makes it transition to \uppLoc{Initial}.
+\begin{figure}[h]
+    \centering
+    \includegraphics[width=1\linewidth]{sprint4/simplified-client.png}
+    \caption{The Client template of the simplified \uppaal model.}
+    \label{fig:sprint4-uppaal-simplified-client}
+\end{figure}
+
+\noindent
+We will now take a look at the more extensive \uppaal model and describe the templates in it.
```

Comment:
```suggestion
The more extensive \uppaal model will now be discussed and the templates in it will be described, starting with the system declarations of the model.
```